src/main/java/com/arpnetworking/http/Routes.java

```diff
@@ -135,14 +149,27 @@ public Routes(
                                                         "{\"status\":\""
                                                                 + (isHealthy ? HEALTHY_STATE : UNHEALTHY_STATE)
                                                                 + "\"}")));
-            } else if (_statusPath.equals(request.getUri().path())) {
+            } else if (_statusPath.equals(path)) {
                 return CompletableFuture.completedFuture(
                         HttpResponse.create()
                                 .withStatus(StatusCodes.OK)
                                 .withEntity(JSON_CONTENT_TYPE, ByteString.fromString(STATUS_JSON)));
             }
         }
-        return CompletableFuture.completedFuture(HttpResponse.create().withStatus(404));
+
+        if (path.startsWith(SOURCE_PREFIX)) {
+            final Future<ActorRef> refFuture = _actorSystem.actorSelection("/user" + path)
+                    .resolveOne(FiniteDuration.create(1, TimeUnit.SECONDS));
```

Comment:
And you don't like my magic sleeps in tests, how about magic timeouts in production code? ;)